skflow/io/data_feeder.py

```diff
@@ -183,14 +195,17 @@ def get_feed_dict_fn(self, input_placeholder, output_placeholder):
             from X and y.
         """
         def _feed_dict_fn():
-            inp = np.zeros(self.input_shape, dtype=self.input_dtype)
+            # take random indices
+            batch_indices = self.indices[self.offset: self.offset+self.batch_size]
```

Comment:
It won't be epoch if you try to repeat data. For example, batch_size=512 and you have only 128 features. If it's circular than you will have 512/128=4 epoch instead of 1 epoch.